backend/src/launch.ts

```diff
@@ -1,54 +1,92 @@
+  minimizeWindows: 'windows/minimize',
 }
 
 export const openCMDforWindows = (launchApp: any) => {
+  if (launchApp.pathPutty) return lunchPutty(launchApp.pathPutty, launchApp.command)
 
+  EventBus.emit(EVENTS.notInstalledPutty, { install: false, loading: true })
+  Logger.info('LAUNCH APP', { launchApp })
+  const process = child_process.exec(`DIR /S putty.exe /B`, { cwd: 'C:\\' })
 
-    if (launchApp.pathPutty) return lunchPutty(launchApp.pathPutty, launchApp.command)
-
-    EventBus.emit(EVENTS.notInstalledPutty, {install: false, loading: true})
-    Logger.info("LAUNCH APP", {launchApp})
-    const process = child_process.exec(`DIR /S putty.exe /B`,  {cwd: 'C:\\'});
-
-    process.stdout.on('data', (data: string) => {
-      Logger.info(`stdout: ${data}`)
-      Logger.info(`stdout: ${data.replace('\\putty.exe', '').replace(/\\/g, '\\\\')}`)
-      const cwd = data.replace('\\putty.exe', '').replace(/\\/g, '\\\\').trim()
-
-      EventBus.emit(EVENTS.notInstalledPutty, {install: false, loading: false, pathPutty: cwd})
-
-      lunchPutty(cwd, launchApp.command)
-    });
-    
-    process.stderr.on('data', (data: string) => {
-      Logger.info(`stderr: ${data}`);
-    });
-    
-    process.on('close', (code: any) => {
-      Logger.info(`child process exited with code ${code}`);
-    });
-
-    child_process.exec(`DIR /S putty.exe /B`,  {cwd: 'C:\\'}, (error: any, stdout: any, stderr: any) => {
-        error && Logger.error(`error: ${error}`)
-        Logger.info(`RESULT PUTTY`, {stdout, stderr})
-        if (!stdout.trim()) {
-            EventBus.emit(EVENTS.notInstalledPutty, {install: true, loading: false})
-        }
-    });
+  process.stdout.on('data', (data: string) => {
+    Logger.info(`stdout: ${data}`)
+    Logger.info(`stdout: ${data.replace('\\putty.exe', '').replace(/\\/g, '\\\\')}`)
+    const cwd = data.replace('\\putty.exe', '').replace(/\\/g, '\\\\').trim()
+
+    EventBus.emit(EVENTS.notInstalledPutty, { install: false, loading: false, pathPutty: cwd })
+
+    lunchPutty(cwd, launchApp.command)
+  })
+
+  process.stderr.on('data', (data: string) => {
+    Logger.info(`stderr: ${data}`)
+  })
+
+  process.on('close', (code: any) => {
+    Logger.info(`child process exited with code ${code}`)
+  })
+
+  child_process.exec(`DIR /S putty.exe /B`, { cwd: 'C:\\' }, (error: any, stdout: any, stderr: any) => {
+    error && Logger.error(`error: ${error}`)
+    Logger.info(`RESULT PUTTY`, { stdout, stderr })
+    if (!stdout.trim()) {
+      EventBus.emit(EVENTS.notInstalledPutty, { install: true, loading: false })
+    }
+  })
+}
+
+export const openCMDvnc = (launchApp: any) => {
+  if (launchApp.pathVNC) return launchVNC(launchApp.pathVNC, launchApp.port, launchApp.host, launchApp.username)
+  EventBus.emit(EVENTS.notInstalledVNC, { install: false, loading: true })
+  Logger.info('LAUNCH APP', { launchApp })
+  const process = child_process.exec(`DIR /S vncviewer.exe /B`, { cwd: 'C:\\' })
```

Comment:
This search performed very slowly for me - if there is a faster way that would be great!

Reply:
we had the same issue in SSH, I remember that we were talking and we didn't find any improve there